explorer/testdata/for/for-loop-auto.carbon

```diff
@@ -0,0 +1,25 @@
+// Part of the Carbon Language project, under the Apache License v2.0 with LLVM
```

Comment:
What is this file supposed to test? The file name mentions `auto`, but I don't see `auto` anywhere in the code.